packages/flutter_tools/bin/tool_backend.sh

```diff
@@ -6,4 +6,4 @@
 readonly flutter_bin_dir="${FLUTTER_ROOT}/bin"
 readonly dart_bin_dir="${flutter_bin_dir}/cache/dart-sdk/bin"
 
-exec "${dart_bin_dir}/dart" "${FLUTTER_ROOT}/packages/flutter_tools/bin/tool_backend.dart" "${@:1}"
+exec "${dart_bin_dir}/dart" "${FLUTTER_ROOT}/packages/flutter_tools/bin/tool_backend.dart" linux "${@:1}"
```

Comment:
This looks incorrect. `${@:1}` is a slice of the positional parameters array (`$@`) that contains all parameters except the 0th. Equivalent to `List.sublist(1)` in Dart.
Ref: https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html#Shell-Parameter-Expansion

Here's where we pass those two parameters in the Linux template, for example:
https://github.com/flutter/flutter/blob/efbde443d2a6f810d0ceb6097fc79ed0b30bddce/packages/flutter_tools/templates/app/linux.tmpl/flutter/CMakeLists.txt#L80-L81

Reply:
So I guess in my case ${FLUTTER_TARGET_PLATFORM} is empty. I would add quotes on the parameters and see

Reply:
Sounds like you are taking a project that has been build on master/dev after https://github.com/flutter/flutter/pull/61221 and trying to build it on stable. Building Flutter projects that have been migrated with older versions than the migration is not, and has never been supported.